src/command/backup/file.c

```diff
@@ -285,62 +285,85 @@ backupFile(
                     // Add size filter last to calculate repo size
                     ioFilterGroupAdd(ioReadFilterGroup(storageReadIo(read)), ioSizeNew());
 
-                    // Open the source and destination and copy the file
+                    // Open the source
                     if (ioReadOpen(storageReadIo(read)))
                     {
-                        // Setup the repo file for write. There is no need to write the file atomically (e.g. via a temp file on
-                        // Posix) because checksums are tested on resume after a failed backup. The path does not need to be synced
-                        // for each file because all paths are synced at the end of the backup. It needs to be created in the prior
-                        // context because it will live longer than a single loop when more than one file is being written.
-                        if (write == NULL)
+                        Buffer *const buffer = bufNew(ioBufferSize());
+
+                        do
                         {
-                            MEM_CONTEXT_PRIOR_BEGIN()
+                            ioRead(storageReadIo(read), buffer);
+                            // open destination and copy the file only if read something or not bundling
```

Comment:
I suggest to replace this comment with "Add file to bundle only if the file size is not 0". A comment should explain things that are not obvious.

Reply:
done